incubator/hnc/api/v1alpha1/hierarchical_namespace.go

```diff
@@ -19,8 +19,12 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// Constant for the hierarchicalnamespaces resource type.
-const HierarchicalNamespaces string = "hierarchicalnamespaces"
+// Constants for the HierarchicalNamespace resource, kind and API version.
+const (
+	HierarchicalNamespaces           = "hierarchicalnamespaces"
+	HierarchicalNamespacesKind       = "HierarchicalNamespace"
+	HierarchicalNamespacesAPIVersion = "hnc.x-k8s.io/v1alpha1"
```

Comment:
I think the "hnc.x-k8s.io" constant is defined somewhere else, maybe use that here as well? I think you can say something like `foo = MetaGroup + "/v1alpha1"`